crates/core/src/commands/check.rs

```diff
@@ -52,13 +99,17 @@ impl CheckOptions {
     /// # Errors
     ///
     /// If the repository is corrupted
-    pub(crate) fn run<P: ProgressBars, S: Open>(self, repo: &Repository<P, S>) -> RusticResult<()> {
+    pub(crate) fn check<P: ProgressBars, S: Open>(
+        &self,
+        opts: CheckOptions,
```

Comment:
I like that much better, it's a good usage of structs, to pass these inside like this, so it's also easy to `impl Default on CheckOptions` 👍🏽